### PR TITLE
Transformation improvements

### PR DESCRIPTION
**Pull request by tbennun**
- [x] Improve HoistState
- [x] Fix bugs in `nest_state_subgraph`
- [x] Helper functions in `dace.transformation.helpers`
- [x] WarpTiling as a GPU-specialization transformation

### CODE REVIEW

File: dace/codegen/compiler.py [hunk 1]
```diff
@@ -196,21 +200,22 @@ def configure_and_compile(program_folder,
...
-        ' '.join(cmake_link_flags) + ' ' +
+        ' '.join(sorted(cmake_link_flags)) + ' ' +
         (Config.get('compiler', 'linker', 'args') or '')).strip()
     if cmake_link_flags:
         cmake_command.append(
             f'-DCMAKE_SHARED_LINKER_FLAGS="{cmake_link_flags}"')
     cmake_command = ' '.join(cmake_command)
 
-    cmake_filename = os.path.join(build_folder, 'cmake_configure.sh')
+    cmake_filename = os.path.join(build_folder, 'cmake_configure.sh')
```
Comment (orausch) [inline dace/codegen/compiler.py:210]:
```suggestion
    cmake_filename = os.path.join(build_folder, 'cmake_configure.sh')
```

File: dace/transformation/dataflow/map_fusion.py [hunk 2]
```diff
@@ -138,8 +138,8 @@ def can_be_applied(graph, candidate, expr_index, sdfg, strict=False):
         # Create a dict that maps parameters of the first map to those of the
```
Comment (orausch) [inline dace/transformation/dataflow/map_fusion.py:138]: Does this comment need to be updated?

File: dace/transformation/helpers.py [hunk 3]
```diff
@@ -742,3 +747,176 @@ def extract_map_dims(sdfg: SDFG, map_entry: nodes.MapEntry,
...
+            if curscope is scope:
+                return True
+            curscope = curstate.entry_node(curscope)
+        curstate = cursdfg.parent
+        curscope = cursdfg.parent_nsdfg_node
+        cursdfg = cursdfg.parent_sdfg
+    return False
+
+
+def redirect_edge(
```
Comment (orausch) [inline dace/transformation/helpers.py:879]: Could be moved to SDFGState or MultiConnectorEdge

**Review by orausch — APPROVED** [review]

File: dace/transformation/dataflow/map_fusion.py (hunk 2 above)
Comment (tbennun) [reply dace/transformation/dataflow/map_fusion.py:138]: It's still correct, just not literal :)

Comment (tbennun) [reply dace/transformation/dataflow/map_fusion.py:138]: It's still correct, just not literal :)

File: dace/transformation/helpers.py (hunk 3 above)
Comment (tbennun) [reply dace/transformation/helpers.py:879]: it's a transformation helper, but I see what you mean. TODO(later)

**Pull request merged**